### ISSUE TITLE
Question about receptive field

### ISSUE THREAD

**Issue by GingerCohle** [aosokin/os2d#5]
![image](https://user-images.githubusercontent.com/37873318/89012379-b8ce8300-d344-11ea-8159-353cadc98bf5.png)
why the feature map went through and its rf==16，whts does this "feature_map_receptive_field=FeatureMapSize(h=16, w=16) "mean? And in ur concat both transformnet([7,5,5] the rf is15) and baacbone. So how did the rf in bacbone feature extraction is 16?

**Comment by aosokin** [comment]
Hi @GingerCohle, thanks for your interest.
This is a confusing point in the code: I used class FeatureMapSize to store objects with the width and height but which are not feature maps.
Here the FeatureMapSize objects are used to store the stride of the feature extractor and the effective receptive field of each position in the feature map. The stride is the actual stride of the used part of ResNet. The effective receptive field is smaller than the true receptive field - it ignores padding in 3x3 convolutions. I did it this way to mimic how SSD creates anchor locations.
Best,
Anton
